### PR TITLE
use flatpak's extra-data and apply_extra

### PR DESCRIPTION
**Pull request by cbm755**
~~Not function yet, just WIP~~

### CODE REVIEW

File: org.srb2.SRB2Kart.yaml [hunk 1]
```diff
@@ -47,4 +49,16 @@ modules:
...
+        sha256: 57284de43a2889c0a60ce8fee77df22095fcd31beef6130cbeed1ead0bebbfec
+        filename: srb2kart-v12-Installer.exe
+        size: 200732615
+        commands:
+          - mkdir tmp
+          - cd tmp; 7za x ../srb2kart-v12-Installer.exe
+          - mkdir -p $FLATPAK_TODO_USER_VAR/sth/sth
+          - cp tmp/srb2.srb $FLATPAK_TODO_USER_VAR/sth/sth
+          - cp tmp/{patch,gfx,textures,chars,maps}.kart $FLATPAK_TODO_USER_VAR/sth/sth
+          - cp tmp/{music,sounds,bonuschars}.kart $FLATPAK_TODO_USER_VAR/sth/sth
```
Comment (cbm755) [inline org.srb2.SRB2Kart.yaml]: @ghisvail Any idea how to actually unpack it?  Do I need to abuse the first-run like your zenity bit does right now?

That seems unpleasant...  I want these commands (something like them) to happen at install time...

Comment (ghisvail) [reply org.srb2.SRB2Kart.yaml]: > I want these commands (something like them) to happen at install time...

Which is what `extra-data` actually provides, together with a script called `apply_extra`. Look for its usage in the flatpak org. I believe repack of Slack or Spotify use it.

Comment (cbm755) [reply org.srb2.SRB2Kart.yaml]: Ah-ha!  This feature could really use better docs.

**Comment by cbm755** [comment]
I cannot figure out how to get the files out of `/app/extra/`: maybe they can just stay there.

But (ab)using the `SRB2WADDIR=/app/extra` doesn't feel quite right: i.e., someone may want to set this to a custom value.  Leave for a future improvement?

**Comment by ghisvail** [comment]
> someone may want to set this to a custom value

Since the app is sandboxed and won't have access to the host home directory, I can't figure why someone would want to do that.

**Comment by cbm755** [comment]
Probably it is better to set an environment variable "globally" rather than in this script: if you know how to do that, please go ahead.

**Comment by ghisvail** [comment]
> Probably it is better to set an environment variable "globally" rather than in this script

Actually, we usually do it in a wrapper script. It is a common pattern (for instance in Electron apps).

**Comment by ghisvail** [comment]
I managed to get something working with `org.srb2.SRB2`. I have synchronized both code bases. The model files are still missing but is just a matter of finding the right patterns in the `7z` and `unzip` calls.

**Comment by ghisvail** [comment]
> finding the right patterns in the 7z and unzip calls.

Done :-)

**Comment by cbm755** [comment]
nice!

**Pull request closed without merging**